101-databricks-workspace-with-customer-managed-keys/azuredeploy.json

```diff
@@ -0,0 +1,212 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "workspaceName": {
+      "type": "string",
+      "metadata": {
+        "description": "The name of the Azure Databricks workspace to create."
+      }
+    },
+    "pricingTier": {
+      "type": "string",
+      "defaultValue": "premium",
+      "allowedValues": [
+        "standard",
+        "premium"
+      ],
+      "metadata": {
+        "description": "The pricing tier of workspace."
+      }
+    },
+    "location": {
+      "type": "string",
+      "allowedValues": [
+        "westus",
+        "eastus2",
+        "westeurope",
+        "northeurope",
+        "eastus",
+        "southeastasia",
+        "eastasia",
+        "southcentralus",
+        "northcentralus",
+        "westus2",
+        "centralus",
+        "ukwest",
+        "uksouth",
+        "australiaeast",
+        "australiasoutheast",
+        "australiacentral",
+        "australiacentral2",
+        "japaneast",
+        "japanwest",
+        "canadacentral",
+        "canadaeast",
+        "centralindia",
+        "southindia",
+        "westindia",
+        "koreacentral",
+        "southafricanorth",
+        "uaenorth",
+        "francecentral",
+        "brazilsouth",
+        "switzerlandnorth"
+      ],
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "keyVaultName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault name."
+      }
+    },
+    "keyName": {
+      "type": "string",
+      "metadata": {
+        "description": "The Azure Key Vault encryption key name."
+      }
+    }
+  },
+  "variables": {
+    "managedResourceGroupName": "[concat('databricks-rg-', parameters('workspaceName'), '-', uniqueString(parameters('workspaceName'), resourceGroup().id))]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Databricks/workspaces",
+      "apiVersion": "2018-04-01",
+      "name": "[parameters('workspaceName')]",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "[parameters('pricingTier')]"
+      },
+      "properties": {
+        "managedResourceGroupId": "[subscriptionResourceId('Microsoft.Resources/resourceGroups', variables('managedResourceGroupName'))]",
+        "parameters": {
+          "prepareEncryption": {
+            "value": true
+          }
+        }
+      }
+    },
+    {
```

Comment:
if you're going to create the vault in the template (instead of using an existing one) then remove the nested deployment (it's not needed)